llvm/include/llvm/IR/ModuleSummaryIndex.h

```diff
@@ -1272,6 +1272,12 @@ using ModulePathStringTableTy = StringMap<ModuleHash>;
 /// a particular module, and provide efficient access to their summary.
 using GVSummaryMapTy = DenseMap<GlobalValue::GUID, GlobalValueSummary *>;
 
+/// A set of global value summary pointers.
+using GVSummaryPtrSet = SmallPtrSet<GlobalValueSummary *, 4>;
+
+/// The key is module path, and value is a set of global value summary pointers.
```

Comment:
Good question. I think it makes sense to use a set of pointers as suggested, given `ModuleToGVSummaryPtrSet` is computed using pointers as opposed to GUIDs. Going to make the change.

While it's possible to have {mod_A, GUID_foo} and {mod_B, GUID_foo} entries because of comdats, each `GlobalValueSummary*` will be associated with exactly one module.

Reply:
Done. Going to update the subsequent patch after updating this PR.

Reply:
Ah got it, thanks for the clarification!